File: 2025-09-11_maximum_average_subarray_I.java
// Problem: Maximum Average Subarray I
// Link: https://leetcode.com/problems/maximum-average-subarray-i/description/?envType=study-plan-v2&envId=leetcode-75

class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int n=nums.length;

        //calculate teh sum of the first k elements
        long windowSum=0;
        for(int i=0;i<k;i++){
            windowSum+=nums[i];
        }

        long maxSum=windowSum;

        //slide the window
        for(int i=k;i<n;i++){
            windowSum+=nums[i]-nums[i-k];
            maxSum=Math.max(maxSum, windowSum);
        }
        return(double)maxSum/k;
    }
}

File: 2025-09-11_maximum_average_subarray_I.py
// Problem: Maximum Average Subarray I
// Link: https://leetcode.com/problems/maximum-average-subarray-i/description/?envType=study-plan-v2&envId=leetcode-75

class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """

        window_sum=sum(nums[:k])
        max_sum=window_sum

        for i in range(k,len(nums)):
            window_sum+=nums[i]-nums[i-k]
            max_sum=max(max_sum,window_sum)

        return float(max_sum)/k
        
